lib/repl.js

```diff
@@ -415,9 +416,11 @@ function REPLServer(prompt,
   });
 
   var sawSIGINT = false;
+  var sawCtrlD = false;
```

Comment:
Yes, when `turnOffEditorMode` is called, it sets `self.editorMode` to `false`. Why can't that itself be used?